### PR TITLE
nit based on comments

### PR DESCRIPTION
**Pull request by godmodegalactus**

### CODE REVIEW

File: proposals/0016-simd-program-rebatable-account-write-fees.md [hunk 1]
```diff
@@ -0,0 +1,566 @@
...
+
+## Detailed Design
+
+PRAW fees enable dapp developer to decide who to rebate and who to penalize
+through runtime features. In particular, programs needs to be able to penalize,
+even if the application is not CPI'd to.
+
+The checks on the PRAW fees will be taken care of by Solana runtime. Total PRAW
+fees paid will be included in the transaction through a special
+`SetProgramRebatableAccountWriteFees` similar to the `ComputeBudget` program so
```
Comment (mschneider) [inline proposals/0016-simd-program-rebatable-account-write-fees.md:122]: This should be a generic program that the computer budget can later move to

Comment (mschneider) [reply proposals/0016-simd-program-rebatable-account-write-fees.md:122]: TVL interface needs to be mentioned

File: proposals/0016-simd-program-rebatable-account-write-fees.md [hunk 2]
```diff
@@ -0,0 +1,566 @@
...
+headers in the transaction, we take the first occurrence and ignore the others.
+
+```
+enum TransactionHeaderParameter {
+  // existing compute budget parameters
+  SetComputeUnitLimit(u32),
+  SetComputeUnitPrice(u64),
+  SetLoadedAccountsDataSizeLimit(u32),
+  // Limit on maximum of PRAW fees to be paid
+  // Fees is set in milli lamports per CU
```
Comment (tao-stones) [inline proposals/0016-simd-program-rebatable-account-write-fees.md:156]: given what we learned from Priority fee was set at micro-lamports, is "milli lamports" a clear choice for longer-term?

File: proposals/0016-simd-program-rebatable-account-write-fees.md [hunk 3]
```diff
@@ -0,0 +1,566 @@
...
+  SetProgramRebatableAccountWriteFees(u64),
+}
+
+enum TransactionHeaderInstruction {
+  parameter: Vec<TransactionHeaderParameter>
```
Comment (tao-stones) [inline proposals/0016-simd-program-rebatable-account-write-fees.md:161]: looks reasonable to me. tbh, not sure if need to provide vec's length as part of parameter tho. Need to dig more later.

Comment (godmodegalactus) [reply proposals/0016-simd-program-rebatable-account-write-fees.md:161]: Let me do some test, creating a project smart contract and using this type.

File: proposals/0016-simd-program-rebatable-account-write-fees.md (hunk 2 above)
Comment (godmodegalactus) [reply proposals/0016-simd-program-rebatable-account-write-fees.md:156]: We found u32 as a good choice as with u32::MAX in millilamports per CU you can update the accounts with less SOL, and really punish malicious actors if they are write locking accounts. But if you think u64 is more coherent with microlamports I can update the proposal accordingly.

Comment (mschneider) [reply proposals/0016-simd-program-rebatable-account-write-fees.md:156]: main argument against u64 would be bandwidth / storage impact on ledger, for the use-cases we evaluated u32 makes a lot of sense

File: proposals/0016-simd-program-rebatable-account-write-fees.md (hunk 3 above)
Comment (mschneider) [reply proposals/0016-simd-program-rebatable-account-write-fees.md:161]: tvl parsing doesn't require vec, TransactionMessage itself encodes length of data bytes, so actual tvl values can be parsed without any tag / length at the beginning of data.
we just need to define a program id for the transaction header program, no instruction enum or vec.

File: proposals/0016-simd-program-rebatable-account-write-fees.md (hunk 2 above)
Comment (tao-stones) [reply proposals/0016-simd-program-rebatable-account-write-fees.md:156]: I was thinking lamport per CU 😜

**Pull request merged**